src/wallet/rpcwallet.cpp

```diff
@@ -164,8 +164,7 @@ UniValue getnewaddress(const JSONRPCRequest& request)
 
     OutputType output_type = pwallet->m_default_address_type;
     if (!request.params[1].isNull()) {
-        output_type = ParseOutputType(request.params[1].get_str(), pwallet->m_default_address_type);
-        if (output_type == OutputType::NONE) {
+        if (!ParseOutputType(request.params[1].get_str(), output_type)) {
             throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, strprintf("Unknown address type '%s'", request.params[1].get_str()));
```

Comment:
Mind adding a test for an empty string?

Reply:
> Mind adding a test for an empty string?

Added tests in 3454f7bf3b4f1ae041c549dbe9e1e9fae3a8b4aa